src/site/content/en/blog/intersectionobserver-v2/index.md

```diff
@@ -0,0 +1,240 @@
+  detect if the intersecting element was visible at the time of intersection.
+date: 2019-02-15
+updated: 2021-02-26
+tags:
+  - blog
+  - intersectionobserver
+hero: image/8WbTDNrhLsU0El80frMBGE4eMCD3/wUCO2CgK5UOz6URpXEtd.jpeg
+alt: Eye of a cat looking through a gap in a fence.
+---
+
+Intersection Observer&nbsp;v1 is one of those APIs that's probably universally loved, and, now that
+[Safari supports it](https://webkit.org/blog/8582/intersectionobserver-in-webkit/) as well,
+it's also finally universally usable in all major browsers. For a quick refresher of the API,
+I recommend watching [Surma](../../../resources/contributors/surma)'s
+[Supercharged Microtip](https://www.youtube.com/embed/kW_atFXMG98) on Intersection
+Observer&nbsp;v1—also embedded below for your viewing pleasure—or reading Surma's in-depth
+[article](../../2016/04/intersectionobserver).
+People have used Intersection Observer&nbsp;v1 for a wide range of use cases like
+[lazy loading of images and videos](../../../fundamentals/performance/lazy-loading-guidance/images-and-video/),
+[being notified when elements reach `position: sticky`](../../2017/09/sticky-headers),
+[fire analytics events](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/visibility-manager.js),
+and many more.
+
+{% YouTube 'kW_atFXMG98' %}
+
+For the full details, check out the
+[Intersection Observer docs on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API),
+but as a short reminder, this is what the Intersection Observer&nbsp;v1 API looks like in the most
+basic case:
+
+```js
+const onIntersection = (entries) => {
+  for (const entry of entries) {
+    if (entry.isIntersecting) {
+      console.log(entry);
+    }
+  }
+};
+
+const observer = new IntersectionObserver(onIntersection);
+observer.observe(document.querySelector('#some-target'));
+```
+
+## What's challenging with Intersection Observer&nbsp;v1?
+
+To be clear, Intersection Observer&nbsp;v1 is great, but it's not perfect. There are
+some corner cases where the API falls short. Let's have a closer look!
+The Intersection Observer&nbsp;v1 API can perfectly tell you when an element is scrolled into the
+window's viewport, but it *doesn't* tell you whether the element is covered
+by any other page content (that is, when the element is occluded) or whether
+the element's visual display has been modified by visual effects like `transform`, `opacity`,
+`filter`, etc., which *effectively* can make it invisible.
+
+Now while for an element in the top-level document this information can be determined by analyzing
+the DOM via JavaScript, for example via
+[`DocumentOrShadowRoot.elementFromPoint()`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/elementFromPoint)
+and then digging deeper, the same information cannot be obtained if the element in question is
+located in a third-party iframe.
```

Comment:
I don't understand this sentence at all.

Reply:
Split up into two sentences.